### ISSUE TITLE
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1

### ISSUE THREAD

**Issue by Ethan7011** [PathwayAndDataAnalysis/causalpath#23]
Hi,
When I carried out CausalPath with the example data, the following error occurs:
```
directory = sample
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at org.panda.resource.HGNC.lambda$load$0(HGNC.java:164)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.SliceOps$1$1.accept(SliceOps.java:199)
        at java.base/java.nio.file.FileChannelLinesSpliterator.forEachRemaining(FileChannelLinesSpliterator.java:117)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at org.panda.resource.HGNC.load(HGNC.java:162)
        at org.panda.resource.HGNC.load(HGNC.java:149)
        at org.panda.resource.FileServer.init(FileServer.java:53)
        at org.panda.resource.FileServer.<init>(FileServer.java:37)
        at org.panda.resource.HGNC.<init>(HGNC.java:17)
        at org.panda.resource.HGNC.get(HGNC.java:31)
        at org.panda.resource.siteeffect.PhosphoSitePlus.lambda$load$1(PhosphoSitePlus.java:76)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.nio.file.FileChannelLinesSpliterator.forEachRemaining(FileChannelLinesSpliterator.java:117)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at org.panda.resource.siteeffect.PhosphoSitePlus.load(PhosphoSitePlus.java:77)
        at org.panda.resource.FileServer.init(FileServer.java:53)
        at org.panda.resource.FileServer.<init>(FileServer.java:37)
        at org.panda.resource.siteeffect.SiteEffectServer.<init>(SiteEffectServer.java:17)
        at org.panda.resource.siteeffect.PhosphoSitePlus.<init>(PhosphoSitePlus.java:17)
        at org.panda.resource.siteeffect.PhosphoSitePlus.get(PhosphoSitePlus.java:25)
        at org.panda.resource.tcga.ProteomicsFileRow.<init>(ProteomicsFileRow.java:42)
        at org.panda.causalpath.resource.ProteomicsFileReader.readAnnotation(ProteomicsFileReader.java:83)
        at org.panda.causalpath.run.CausalPath.run(CausalPath.java:354)
        at org.panda.causalpath.run.CausalPath.main(CausalPath.java:286)
```

The causalpath.jar file was downloaded from https://github.com/PathwayAndDataAnalysis/causalpath/releases, and my java version  is as follows：
openjdk version "1.8.0_412"
OpenJDK Runtime Environment (Zulu 8.78.0.19-CA-linux64) (build 1.8.0_412-b08)
OpenJDK 64-Bit Server VM (Zulu 8.78.0.19-CA-linux64) (build 25.412-b08, mixed mode)

When I checked the .panda folder, I found the hgnc.txt as follows：

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>301 Moved Permanently</title>
</head><body>
<h1>Moved Permanently</h1>
<p>The document has moved <a href="http://www.genenames.org/cgi-bin/download/?col=gd_hgnc_id&amp;col=gd_app_sym&amp;col=gd_prev_sym&amp;col=gd_aliases&amp;col=gd_pub_chrom_map&amp;col=family.name&amp;col=md_prot_id&amp;status=Approved&amp;status_opt=2&amp;where=&amp;order_by=gd_app_sym_sort&amp;format=text&amp;limit=&amp;hgnc_dbtag=on&amp;submit=submit">here</a>.</p>
<hr>
<address>Apache/2.4.57 (Ubuntu) Server at www.genenames.org Port 80</address>
</body></html>
```

Please let me know what I should do.

Thanks
Ethan

**Comment by kunpengLiLkp** [comment]
the hgnc.txt  cannot be obtained，Web page move Permanently.I have found it and left a message to the author. hope the author will update it as soon as possible.

**Comment by ozgunbabur** [comment]
Thanks for reporting this. We just updated the HGNC URL in the codebase. The causalpath.jar file in the releases does not have this update though. Please follow the instructions on the front page to generate a new jar file. We will make a new release soon.

**Comment by AlexBRHC** [comment]
Hi,
any updates on when an updated release is to be expected? having the same issue.

Thanks in advance,
Alex

**Comment by ozgunbabur** [comment]
Hi Alex,

The error you encountered is different than the previously reported one. Thanks for letting us know. We just fixed it in the latest release (v1.2.1). Please download and try again.

Ozgun

**Comment by AlexBRHC** [comment]
Thanks for the quick fix! 

Best,Alex
